parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -696,11 +730,15 @@ protected <R> void supervisorLoop(Function<PollContextInternal<K, V>, List<R>> u
                     controlLoop(userFunctionWrapped, callback);
                 } catch (InterruptedException e) {
                     log.debug("Control loop interrupted, closing");
-                    doClose(DrainingCloseable.DEFAULT_TIMEOUT);
+                    Thread.interrupted(); //clear interrupted flag as during close need to acquire commit locks and interrupted flag will cause it to throw another interrupted exception.
+                    doClose(shutdownTimeout);
                 } catch (Exception e) {
+                    if (Thread.interrupted()) { //clear interrupted flag
```

Comment:
@jhollandus, @eddyv - can you have a look at this - seems this control thread can be interrupted but we need it to not have interrupted flag set when doing doClose() as there is some code that is sensitive to that flag..

Reply:
From what I understand, since we are actually actively trying to perform some actions as a result of an interrupt, it seems fine to me to clear the flag to perform some work. We may want to consider re-throwing the flag after we are done with the close? See [stackoverflow question](https://stackoverflow.com/questions/10401947/methods-that-clear-the-thread-interrupt-flag)?

Reply:
Dont really see a point resetting the interrupt flag - the only way to reset it is to interrupt the thread again - and that will lead to InterruptedException (unless the thread is blocked etc). But it is a shutdown process - so after we are done with it - the PC - including that control-thread is shut down / closed.

Reply:
Yeah - the next thing after finishing doClose - the while(state!=CLOSED) loop will finish (as we transition to CLOSED state during the doClose call) and the Future will be set / done. So the control thread will be done at that point.

Reply:
Great